gravitee-apim-portal-webui-next/src/app/catalog/catalog.component.ts

```diff
@@ -92,19 +98,33 @@ export class CatalogComponent {
   }
 
   public onFilterSelection(event: string) {
-    this.selectedFilter = event;
+    this.filter = event;
+    this.router.navigate([''], {
+      relativeTo: this.route,
+      queryParams: {
+        filter: this.filter === 'all' ? '' : this.filter,
+        query: this.query,
+      },
+    });
     this.page$.next(1);
   }
 
   public onSearchResults(searchInput: string) {
-    this.searchInput = searchInput;
+    this.query = searchInput;
+    this.router.navigate([''], {
+      relativeTo: this.route,
+      queryParams: {
+        filter: this.filter,
+        query: this.query,
+      },
+    });
```

Comment:
Same here: the navigate will update the query, no need to set it here.

Reply:
Done